ratpack-core/src/test/groovy/ratpack/http/client/HttpClientSmokeSpec.groovy

```diff
@@ -291,6 +292,27 @@ class HttpClientSmokeSpec extends HttpClientSpec {
     response.statusCode == 500
   }
 
+  def "can set connect timeout"() {
+    setup:
+    def nonRoutableIp = '192.168.0.0'
+
+    when:
+    handlers {
+      get { HttpClient httpClient ->
+        httpClient.get("http://$nonRoutableIp".toURI()) {
+          it.connectTimeout(Duration.ofMillis(20))
+        } onError {
+          render it.class.name
```

Comment:
Do you want me to make this change Dan? I checked for the specific exception as I wanted to make sure it was a connection timeout not any other exception as I was initially getting tripped up by the read timeout (which I believe defaults to 30 seconds) when I was connecting to a listening address.